### ISSUE TITLE
Periodic BC does not work for monomial variables

### ISSUE THREAD

**Issue by lw4992** [idaholab/moose#5332]
Periodic BC did't work for monomial variables in my DG simulation. In the `test` directory, the nodal variables were tested. Is it an issure here?

**Comment by jwpeterson** [comment]
I don't think this works, at least in the way that MOOSE imposes periodic BCs, since monomial variables don't have nodal degrees of freedom.  I think I can see how it might work, though, by pairing an "inflow" boundary on the left with an equivalent "outflow" boundary on the right... This is not currently implemented, though.

**Comment by permcody** [comment]
I don't believe anyone has ever tried this so I can see why there may be a
gap. It shouldn't be too bad to fix as the methods for finding periodic
neighbors does work, we probably just don't have that logic added in the DG
loops inside of MOOSE. I'll open a ticket to get this added.

Cody

**Comment by permcody** [comment]
Just curious what kind of simulation are you doing where you need Periodic
BCs with DG?

**Comment by lw4992** [comment]
I'm tring moose for CFD simulation. There is an interesting case called Taylor-Green vortex problem. The initial condition looks like 
![tgv1](https://cloud.githubusercontent.com/assets/7278509/8457494/46fd1628-2044-11e5-965c-a575b402eaa5.png)

Solve it fllowing Navier-Stokes equations, the flow structure will break down step by step , 1 to 2, 2 to 4, 4 to 8...., finally, turbulence occurs. In this case, periodic bcs were required  in x, y, z direction.
![tgv2](https://cloud.githubusercontent.com/assets/7278509/8457499/4e5e4194-2044-11e5-852a-72bb21ed111b.png)

**Comment by lw4992** [comment]
Thank @jwpeterson. In fact, I have tried to implement the strategy as you said. but moose is ingenious for me, I'm failed.
